src/hooks/useFacilitiesRtData.tsx

```diff
@@ -0,0 +1,39 @@
+import { useContext, useEffect } from "react";
+
+import { getRtDataForFacility } from "../infection-model/rt";
+import { FacilityContext } from "../page-multi-facility/FacilityContext";
+import { Facilities } from "../page-multi-facility/types";
+
+const useFacilitiesRtData = (facilities: Facilities | null, useRt = false) => {
+  const { rtData, dispatchRtData } = useContext(FacilityContext);
+
+  async function getRtDataForFacilities(facilities: Facilities) {
+    return await Promise.all(
+      [...facilities].map(async (facility) => {
+        // don't fetch data if we already have it
+        if (rtData && rtData[facility.id]) return;
```

Comment:
Yeah that makes sense, maybe we could extend this object to be closer to a `{loading, failed, data } setup so that we can check whether the request failed

Reply:
yeah works for me ... this undefined vs null thing has caused a problem literally everywhere anyone has tried to work with it (including myself immediately after I implemented it) so I think your suggestion is a good one as well